service/state/core_access.go

```diff
@@ -0,0 +1,93 @@
+package state
+
+import (
+	"context"
+	"fmt"
+
+	sdk_tx "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/tendermint/spm/cosmoscmd"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/celestiaorg/celestia-app/app"
+	apptypes "github.com/celestiaorg/celestia-app/x/payment/types"
+)
+
+// CoreAccessor implements Accessor over an RPC connection
+// with a celestia-core node.
+type CoreAccessor struct {
+	signer *apptypes.KeyringSigner
+	encCfg cosmoscmd.EncodingConfig
+
+	coreEndpoint string
+	coreConn     *grpc.ClientConn
+}
+
+// NewCoreAccessor dials the given celestia-core endpoint and
+// constructs and returns a new CoreAccessor with the active
+// connection.
+func NewCoreAccessor(
+	signer *apptypes.KeyringSigner,
+	encCfg cosmoscmd.EncodingConfig,
+	endpoint string,
+) (*CoreAccessor, error) {
+	return &CoreAccessor{
+		signer:       signer,
+		encCfg:       encCfg,
+		coreEndpoint: endpoint,
+	}, nil
+}
+
+func (ca *CoreAccessor) Start(_ context.Context) error {
+	if ca.coreConn != nil {
+		return fmt.Errorf("core-access: already connected to core endpoint")
+	}
+	// dial given celestia-core endpoint
+	client, err := grpc.Dial(ca.coreEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
Dial will send error immediately If there even a small chance that server could start later than client. I'd suggest to add [WithBlock](https://pkg.go.dev/google.golang.org/grpc#WithBlock) option + context with Timeout. In this case you need [DialContext](https://pkg.go.dev/google.golang.org/grpc#DialContext) instead of Dial

Reply:
Resolved?